File: DataStructure/DifferentialArray/leetcode2772/DA_Greedy_2772.cpp
// class Solution 
// {
// public:
//     bool checkArray(vector<int>& nums, int k) 
//     {
//         const int n=nums.size();
//         vector<int>diff(n+1);

//         int i,sum_diff=0;
//         for(i=0;i<n;++i)
//         {
//             sum_diff+=diff[i];
//             int x=nums[i];

//             x+=sum_diff;
//             if(!x)continue;

//             if(x<0 || i+k>n)return false;
//             sum_diff-=x;
//             diff[i+k]+=x;
//         }


//         return true;
//     }
// };

File: DataStructure/DifferentialArray/leetcode2772/DA_Greedy_2772.go
// func checkArray(nums []int, k int) bool {
//     n:=len(nums)
//     diff:=make([]int,n+1)

//     sum_diff:=0

//     for i,x := range nums{
//         sum_diff+=diff[i]
//         x+=sum_diff

//         if x==0{
//             continue
//         }
//         if x<0 || i+k>n{
//             return false
//         }

//         sum_diff-=x

//         diff[i+k]+=x
//     }

//     return true
// }